src/pkgcheck/addons/git.py

```diff
@@ -79,14 +76,37 @@ def __init__(self, *args, commit):
         self.commit = commit
 
 
+class GitSafeDirectory:
```

Comment:
[suggestion] This is meant to be used to override whatever configuration the user has, maybe a better name is `GitGlobalConfig` or something to that effect? Or just `GitConfig`.